ios/Private/SentryOptions+RNOptions.mm

```diff
@@ -0,0 +1,210 @@
+#import "SentryOptions.h"
+#import "SentryDsn.h"
+#import "SentryLog.h"
+#import "SentryMeta.h"
+#import "SentrySdkInfo.h"
+#import "SentryOptions+RNOptions.h"
+
+@implementation SentryOptions (RNOptions)
+
+
+- (_Nullable instancetype)initWithRNOptions:(NSDictionary<NSString *, id> *)options
```

Comment:
H: Maybe we should do as we do on Flutter, which is calling the properties directly
https://github.com/getsentry/sentry-dart/blob/3f23617e3346caa128c5f27b9c935de4974c3067/flutter/ios/Classes/SentryFlutterPluginApple.swift#L248-L318
So the first question to be answered here is why did we remove the init with Map from Cocoa? So we can consider the trade-offs before going ahead.

Reply:
The init was removed from cocoa mainly because it was not used by anyone (besides us), but because it was present it needed to be maintained.